framework/src/base/AppFactory.C

```diff
@@ -87,5 +88,21 @@ AppFactory::createShared(const std::string & app_type,
   command_line->addCommandLineOptionsFromParams(parameters);
   command_line->populateInputParams(parameters);
 
-  return _name_to_build_info[app_type]->build(parameters);
+  build_info->_app_creation_count[app_type]++;
+
+  return build_info->build(parameters);
+}
+
+size_t
+AppFactory::createdAppCount(const std::string & app_type) const
+{
+  // Error if the application type is not located
+  const auto it = _name_to_build_info.find(app_type);
+  if (it == _name_to_build_info.end())
+    mooseError("Object '" + app_type + "' was not registered.");
+  auto & build_info = _name_to_build_info.at(app_type);
```

Comment:
The advantage of `find` vs `count` is that with `it` already in hand you do not need to do this lookup again

Reply:
Good point!